main/EventSystem.cpp

```diff
@@ -2442,27 +2440,28 @@ void CEventSystem::ExportDomoticzDataToLua(lua_State *lua_state, uint64_t device
 			_hardwareNames[ID] = tlist;
 		}
 	}
+	
+	Json::Value tempjson;
+	m_webservers.GetJSonDevices(tempjson, "true", "", "dzvID", "", "", "", true, false, false, 0, "");
```

Comment:
Agreed however, some of those fields are derived fields like counter results. I think that when  you change some domoticz settings the results of those fields could change as well. Of course they will be updated in the next device-update cycle with the new (formatted) values so that may not be an issue.
But besides this, this PR makes it a whole lot more efficient than what is currently in master so I think this is still a good iteration. 
Next step would be perhaps to add those fields and perhaps look again at @jvandenbroek (retracted) pull request to make better use of multi-core systems when handling events.